services/issue/project.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package issue
+
+import (
+	"errors"
+
+	issues_model "code.gitea.io/gitea/models/issues"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/log"
+)
+
+func CloseIssue(ctx *context.Context, issueID int64) error {
+	issue, err := issues_model.GetIssueByID(ctx, issueID)
+	if err != nil {
+		return errors.New("failed getting issue")
+	}
+
+	if err := ChangeStatus(ctx, issue, ctx.Doer, "", true); err != nil {
```

Comment:
you could also remove `if` nesting here by checking if err is nil and moving the status change function call prior to the `if`, and then return in the if, and after the if, continue with the remaining logic.